src/render_engine/vulkan/vk_structs.hpp

```diff
@@ -7,38 +7,73 @@
 
 #pragma once
 
-#include <vk_mem_alloc.h>
-#include <vulkan/vulkan.hpp>
-
-#include "nova_renderer/rhi_types.hpp"
+#include <vma/vk_mem_alloc.h>
+#include <vulkan/vulkan.h>
+#include <nova_renderer/rhi_types.hpp>
 
+#ifdef ENABLE_VULKAN
 namespace nova::renderer::rhi {
+
     struct VulkanDeviceMemory : DeviceMemory {
-        VkDeviceMemory memory;
+        VmaAllocationInfo info{};
```

Comment:
No

The whole point of DeviceMemory is to enable Nova to manage its own gpu memory. DirectX has no equivalent for VMA, so we could have to do custom memory stuff for that backend. I realized that what Nova needed from GPU memory wasn't super complicated, so I decided to manually do memory management for Vulkan as well

Much of the Vulkan render engine does still use VMA, yes, but only because this branch is very incomplete. VMA will be completely removed before my bra ch is merged into master